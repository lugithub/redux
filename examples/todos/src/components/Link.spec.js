import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import TestRenderer from 'react-test-renderer'
import reducer from '../reducers'
import Link from './Link'
import ReactTestUtils from 'react-dom/test-utils';
import { findRenderedComponentWithType } from 'react-dom/test-utils';
//const store = createStore(reducer)

describe('Link component', () => {
  it('should handle onClick', () => {
    // const tree = TestRenderer.create(
    //   <Link active={false} onClick={e => e}>children</Link>
    // ).toTree();
    const tree = Link({
      active: false,
      children: 'children',
      onClick: console.log,
    });
    const element = findRenderedComponentWithType(tree, 'a');
    ReactTestUtils.Simulate.click(element);

  });
})
