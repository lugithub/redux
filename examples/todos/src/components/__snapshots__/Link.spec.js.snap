// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should handle onClick 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "active": false,
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "href": "#",
      "onClick": [Function],
    },
    "rendered": null,
    "type": "a",
  },
  "type": [Function],
}
`;
